### PR TITLE
Re-format of namespace, update lang version & SDK and improve US852 IFF models

### PR DESCRIPTION
**Pull request by Adanlink**
Lang version to C# 12
SDK to .NET 8
Simplified namespace usage

Improvements to the US852 IFF models:
- IFFCommon -> Added tiki releated data and defined missing types and flags.
- BallUs8 -> Improved ball model with missing data separation
- CaddieUs8 -> Normalized with StatsStruct
- Character -> Made unknown those fields that didn't match the prior names (and had unknown usage) and changed RankS field (uint) to ClubSetScale (float)
- ClubSetUs8 -> Added missing final data related to Workshop and custom Pangya Images
- CourseUs8 -> Added missing data (Unknowns, Pars, MinScore and MaxScore), changed CourseFlag (byte) to Difficulty (b4 unknown, b4 difficultyStars) and put proper names to filenames.
- ItemUs8 -> Added (wasn't present in the lib)
- PartUs8 -> Added (wasn't present in the lib)

Part of the improvements in this models have been based on previous work done by Acrissio in its Japanese Season 8 Server Emulator.
The rest are discoveries found by my own work/investigation.

### CODE REVIEW

File: PangLib.IFF/Model/US/Season8/Type/ClubSetWorkshopType.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace PangLib.IFF.Model.US.Season8.Type;
+
+public enum ClubSetWorkshopType : int
+{
+    NotUpgradable = -1,
+    Upgrdable = 0
```
Comment (pixeldesu) [inline PangLib.IFF/Model/US/Season8/Type/ClubSetWorkshopType.cs]: Small typo
```suggestion
    Upgradable = 0
```

File: PangLib.IFF/Model/US/Season8/Type/PartType.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿namespace PangLib.IFF.Model.US.Season8.Type;
+
+public enum PartType : uint
+{
+    Top = 0,
+    Bottom = 1,
+    Hat = 2,
+    Gloves = 3,
+    Shoes = 4,
+    AccesoryOrBase = 5,
```
Comment (pixeldesu) [inline PangLib.IFF/Model/US/Season8/Type/PartType.cs]: Another typo
```suggestion
    AccessoryOrBase = 5,
```

**Review by pixeldesu — CHANGES_REQUESTED** [review]
I tested the new models, and they seem to work fine on my end! 👍 

Marked two things I found while looking over things, just small typos.

But I have another thing that irks me a bit, and is up for discussion. You renamed the `Models` namespace to `Model`. And, according to the [C# framework guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-namespaces) it says "CONSIDER using plural namespace names where appropriate."

Other than that, everything looks fine!

For other Pull Requests I'd please suggest limiting the scope to a single package, even with the namespace adjustments, reviewing this was kinda hard 😄

**Comment by Adanlink** [comment]
> Marked two things I found while looking over things, just small typos.

Will fix them now in a moment 🤣

> But I have another thing that irks me a bit, and is up for discussion. You renamed the `Models` namespace to `Model`. And, according to the [C# framework guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-namespaces) it says "CONSIDER using plural namespace names where appropriate."

Sure!

> For other Pull Requests I'd please suggest limiting the scope to a single package, even with the namespace adjustments, reviewing this was kinda hard 😄

Yes, you are right, I should have made a separate PR just for the adjustments, my fault. Thanks for reviewing it!

**Review by pixeldesu — APPROVED** [review]

**Pull request merged**